core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -210,13 +215,17 @@ class SearchService(
   /** free-form search for classes */
   def searchClasses(query: String, max: Int): List[FqnSymbol] = {
     val fqns = index.searchClasses(query, max)
-    Await.result(db.find(fqns), QUERY_TIMEOUT) take max
+    (Await.result(db.find(fqns), QUERY_TIMEOUT) take max) sortBy { f =>
+      !applyBoost(f.file drop 7)
```

Comment:
in hindsight `applyBoost` is a terrible name, it should be something like `isUserFile`. I'm using the check here to order the `FQN`s we get from the Db, since as far as I could tell there's no guarantee the ordering of fqns that goes into `db.find` is the same that comes out.